3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -10,21 +10,22 @@
 // ! Do not change or remove the next two lines
 const rollDice = require('../../helpers/pokerDiceRoller');
 
-function rollDiceUntil(wantedValue) {
+async function rollDiceUntil(wantedValue) {
   // TODO: rewrite this function using async/await
-  return rollDice().then((value) => {
-    if (value !== wantedValue) {
-      return rollDiceUntil(wantedValue);
-    }
-    return value;
-  });
+  const value = await rollDice();
+  while (value !== wantedValue) {
+    await rollDice();
+  }
```

Comment:
The while loop condition never changes while looping. So unless the initial rollDice() call in line 15 is already producing an ACE, looping will continue until a dice rolls off the table (promise rejected), but never when an ACE is produced (promise resolved).

```suggestion
  let value = await rollDice();
  while (value !== wantedValue) {
    value = await rollDice();
  }
  return value;
```